client/src/containers/Home/Home.js

```diff
@@ -0,0 +1,109 @@
+import React, { Component, cloneElement } from 'react';
+import { browserHistory } from 'react-router';
+import { Jumbotron, Button } from 'react-bootstrap';
+
+const appStyle = {
+  margin: 'auto auto',
+  width: '80%',
+  height: '100%',
+  border: '1px solid black',
+  padding: '7%',
+  textAlign: 'center',
+  background: '#CCC',
+};
+
+export default class extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      room: null,
+      location: null,
+    };
+
+    this.handlers = {
+      addMessageToChatRoom: this.addMessageToChatRoom.bind(this),
+      createChatRoom: this.createChatRoom.bind(this),
+      joinChatRoom: this.joinChatRoom,
+    };
+
+    this.props.route.socket.on('updateMessagesState', room => {
+      this.setState({ room });
+    });
+  }
+
+  componentWillMount() {
+    if (!this.state.location) {
+      browserHistory.push('/roaming');
+    }
+    this.setState({ username: localStorage.getItem('username') });
+    this.updateLocationState();
+    this.interval = setInterval(this.updateLocationState.bind(this), 500);
```

Comment:
Is this better than using [navigator.geolocation.watchposition](https://developer.mozilla.org/en-US/docs/Web/API/Geolocation/watchPosition)?